src/groups/group.jl

```diff
@@ -450,35 +381,31 @@ left or right `conv`ention. The differential transports vectors:
 (\mathrm{d}τ_p)_q : T_q \mathcal{G} → T_{τ_p q} \mathcal{G}\\
 ```
 """
-translate_diff(M::Manifold, p, q, X) = translate_diff(M, p, q, X, LeftAction())
-function translate_diff(M::Manifold, p, q, X, conv::ActionDirection)
-    return translate_diff(M, p, q, X, conv, is_decorator_manifold(M))
-end
-function translate_diff(M::Manifold, p, q, X, conv::ActionDirection, ::Val{true})
-    return translate_diff(M.manifold, p, q, X, conv)
-end
-function translate_diff(M::Manifold, p, q, X, conv::ActionDirection, ::Val{false})
-    return error("translate_diff not implemented on $(typeof(G)) for elements $(typeof(p)) and $(typeof(q)), vector $(typeof(X)), and direction $(typeof(conv))")
+translate_diff(::AbstractGroupManifold, ::Any...)
+@decorator_transparent_signature translate_diff(M::AbstractDecoratorManifold, p, q, X)
+function translate_diff(G::AbstractGroupManifold, p, q, X)
+    return translate_diff(G, p, q, X, LeftAction())
 end
+@decorator_transparent_signature translate_diff(M::AbstractDecoratorManifold, p, q, X, conv::ActionDirection)
 function translate_diff(G::AbstractGroupManifold, p, q, X, conv::ActionDirection)
-    pq = translate(G, p, q, conv)
     Y = allocate_result(G, translate_diff, X, p, q)
     translate_diff!(G, Y, p, q, X, conv)
     return Y
 end
-
-function translate_diff!(M::Manifold, Y, p, q, X)
+@decorator_transparent_signature translate_diff!(M::AbstractDecoratorManifold, Y, p, q, X)
+function translate_diff!(M::AbstractGroupManifold, Y, p, q, X)
     return translate_diff!(M, Y, p, q, X, LeftAction())
 end
-function translate_diff!(M::Manifold, Y, p, q, X, conv::ActionDirection)
-    return translate_diff!(M, Y, p, q, X, conv, is_decorator_manifold(M))
-end
-function translate_diff!(M::Manifold, Y, p, q, X, conv::ActionDirection, ::Val{true})
-    return translate_diff!(M.manifold, Y, p, q, X, conv)
-end
-function translate_diff!(M::Manifold, Y, p, q, X, conv::ActionDirection, ::Val{false})
-    return error("translate_diff! not implemented on $(typeof(M)) for elements $(typeof(Y)), $(typeof(p)) and $(typeof(q)), vector $(typeof(X)), and direction $(typeof(conv))")
-end
+@decorator_transparent_signature translate_diff!(M::AbstractDecoratorManifold, Y, p, q, X, conv::ActionDirection)
+decorator_transparent_dispatch(
+    ::typeof(translate_diff!),
+    M::AbstractGroupManifold,
+    Y,
+    p,
+    q,
+    X,
+    conv,
+) = Val(:intransparent)
```

Comment:
By convention, if the signature takes multiple lines, then use the long function definition i.e. `function foo(...)`